pkg/js/helpers.js

```diff
@@ -1739,6 +1739,77 @@ function CAA_BUILDER(value) {
     return r;
 }
 
+// DKIM_BUILDER takes an object:
+// label: The DNS label for the DKIM record ([selector]._domainkey prefix is added; default: '@')
+// selector: Selector used for the label. e.g. s1 or mail
+// pubkey: Public key (p) to be used for DKIM.
+// keytype: Key type (k). Defaults to 'rsa' if missing (optional)
+// flags: Which types (t) of flags to activate, ie. 'y' and/or 's'. Array, defaults to 's' (optional)
+// hashtypes: Acceptable hash algorithma (h) (optional)
+// servicetypes: Record-applicable service types (optional)
+// note: Note field fo admins. Avoid if possible to keep record length short. (optional)
+// ttl: The time for TTL, integer or string. (default: not defined, using DefaultTTL)
+
+function DKIM_BUILDER(value) {
+    if (!value) {
+        value = {};
+    }
+    kvs = [];
+
+    if (!value.selector) {
+        throw 'DKIM_BUILDER selector cannot be empty';
+    }
+
+    if (!value.pubkey) {
+        throw 'DKIM_BUILDER pubkey cannot be empty';
```

Comment:
Just a small clarification regarding DKIM: although an empty `p=` value in a DKIM record might look invalid at first glance, it is actually allowed by the DKIM specification (RFC 6376, section 3.6.1).

A record like this:

```text
v=DKIM1; p=
```

is valid and signals that the selector should not be trusted. In practice, I use this to **deactivate a DKIM selector** or to **prevent spoofing** on a domain or subdomain by making sure no valid key is published.

This technique is a deliberate choice, especially for domains or subdomains that should not be sending email at all.